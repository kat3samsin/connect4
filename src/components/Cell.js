import React, {Component} from 'react';
import {connect} from 'react-redux';

import { movePlayer, checkWinner } from '../actions/GameActions'

export class Cell extends Component {
  handleClick = (args) => {
    if (!this.props.winner) {
      this.props.move(args);
    }
  };

  render() {
    return (
      <td className={this.props.player === 1 ? 'board-td p1-td' : 'board-td p2-td'}>
        <div 
          className={this.props.value === 1 ? 'player1' : 
                    this.props.value === 2 ? 'player2' : 'init'}
          onClick={this.handleClick.bind(this, {
            rows: this.props.rows, 
            col: this.props.col,
            player: this.props.player,
            board: this.props.board,
            cellsToWin: this.props.cellsToWin
          })}>
        </div>
      </td>
    );
  }
}

function mapDispatchToProps(dispatch) {
  return {
    move: args => {
      dispatch(movePlayer(args));
      dispatch(checkWinner(args));
    }
  }
}
const mapStateToProps = (state) => {
  return state;
}
export default connect(mapStateToProps, mapDispatchToProps)(Cell);
