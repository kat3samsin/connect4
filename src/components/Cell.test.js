import React from 'react';
import { shallow } from 'enzyme';
import { Cell } from './Cell';

describe('Cell', () => {
  it('renders without crashing', () => {
    let wrapper = shallow(<Cell player={1} value={0} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('renders td', () => {
    it('value is 1 > hasClass p1-td', () => {
      let wrapper = shallow(<Cell player={1} value={0} />);
      expect(wrapper.find('td').hasClass('board-td p1-td')).toEqual(true);
    });

    it('value is 2 > hasClass > p1-td', () => {
      let wrapper = shallow(<Cell player={2} value={0} />);
      expect(wrapper.find('td').hasClass('board-td p2-td')).toEqual(true);
    });
  });

  describe('renders div', () => {
    it('value is 1 > hasClass > player1', () => {
      let wrapper = shallow(<Cell player={1} value={1} />);
      expect(wrapper.find('div').hasClass('player1')).toEqual(true);
    });

    it('value is 2 > hasClass > player2', () => {
      let wrapper = shallow(<Cell player={2} value={2} />);
      expect(wrapper.find('div').hasClass('player2')).toEqual(true);
    });

    it('value is 0 > hasClass > init', () => {
      let wrapper = shallow(<Cell player={1} value={0} />);
      expect(wrapper.find('div').hasClass('init')).toEqual(true);
    });
  });

  describe('simulate div click', () => {
    let moveFn, removeFn;

    beforeEach(() => {
      moveFn = jest.fn();
      removeFn = jest.fn();
    });

    afterEach(() => {
      moveFn.mockReset();
    });

    it('col is 1 > should bind col=1', () => {
      let wrapper = shallow(
        <Cell player={1} value={0} col={1} move={moveFn} remove={removeFn}/>
      );
      wrapper.find('div').simulate('click');
      expect(moveFn).toBeCalledWith({
        board: undefined,
        cellsToWin: undefined,
        col: 1,
        player: 1,
        rows: undefined
      });
    });

    it('col is 3 > should bind col=3', () => {
      let wrapper = shallow(
        <Cell player={1} value={0} col={3} move={moveFn} remove={removeFn}/>
      );
      wrapper.find('div').simulate('click');
      expect(moveFn).toBeCalledWith({
        board: undefined,
        cellsToWin: undefined,
        col: 3,
        player: 1,
        rows: undefined
      });
    });
  });
});
