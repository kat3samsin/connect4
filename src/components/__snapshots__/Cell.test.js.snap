// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell renders without crashing 1`] = `ShallowWrapper {}`;
